core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -100,11 +100,13 @@ object ConsumerPerformance {
     var lastReportTime: Long = startMs
     var lastBytesRead = 0L
     var lastMessagesRead = 0L
-    var lastConsumed = System.currentTimeMillis
-    while(messagesRead < count && lastConsumed >= System.currentTimeMillis - timeout) {
+    var lastConsumedTime = System.currentTimeMillis
+
+
+    while(messagesRead < count && System.currentTimeMillis() - lastConsumedTime < timeout) {
```

Comment:
Yes, my mistake - this is definitely (functionally) equivalent.

The thing I was attempting to fix was the `0 throughput` that you are finding when you run it. More on that below.

On the other hand, this makes the intent clearer (to me at least?).